templates/bootstrap3-horizontal/components/quickForm/quickForm.js

```diff
@@ -5,10 +5,38 @@ Template['quickForm_bootstrap3-horizontal'].helpers({
   labelClass: function () {
     return this.atts["label-class"];
   },
+  fieldGroupLabel: function () {
+    var name = this.name;
+
+    // if field group name is of the form XY_abcde where "XY" is a number, remove prefix
+    if (typeof parseInt(name.substr(0,2)) === "number" && name.charAt(2) === "_") {
+      var name = name.substr(3);
+    }
+
+    // if SimpleSchema.defaultLabel is defined, use it
+    if (typeof SimpleSchema.defaultLabel === "function") {
+      return SimpleSchema.defaultLabel(name);
+    } else {
+      // else, just capitalise name
+      return name.charAt(0).toUpperCase() + name.slice(1)
+    }
+  },
+  getFieldsWithNoGroups: function () {
+    // if there are no fields with no groups, don't call afQuickFields at all
+    return this.fieldsWithNoGroups.fields.length ? this.fieldsWithNoGroups : null;
+  },
   quickFieldsAtts: function () {
     // These are the quickForm attributes that we want to forward to
     // the afQuickFields component.
-    return _.pick(this.atts, 'id-prefix', 'input-col-class', 'label-class');
+
+    // get fields for current field group
+    var fieldsForCurrentGroup = this.fields;
+
+    // clone the object to make sure we don't modify the original data context
+    var atts = _.extend({}, Template.parentData(1).atts);
+    
+    atts.fields = fieldsForCurrentGroup
+    return _.pick(atts, 'id-prefix', 'input-col-class', 'label-class');
```

Comment:
This should include `fields` otherwise you're stripping off the property you just added. I think it may also look back up the data context tree for a `fields` attribute, so it might even work without setting `fields` explicitly here. I don't remember the exact details.